### DISCUSSION TITLE
Sanity check-- is this a safe method of cookie-based auth in go?

### DISCUSSION THREAD

**Discussion by 0ui** [pocketbase/pocketbase#5906] [Q&A] [answered]
I've seen a few other cookie authentication threads but nothing that worked exactly as I'm looking for. Just wanted to know if anybody more familiar with the project could verify that this is not an egregiously unsafe way to go about it.

1) On user login, I'm saving a copy of the token in an http cookie
```go
app.OnRecordAfterAuthWithOAuth2Request().Add(func(e *core.RecordAuthWithOAuth2Event) error {
	t, err := tokens.NewRecordAuthToken(app.App, e.Record)
	if err != nil {
		return err
	}
	e.HttpContext.SetCookie(&http.Cookie{
		Name: "Auth-Token", Value: t, HttpOnly: true, Path: "/",
	})
	return nil
})
```

2) I copied the `LoadAuthContext` middleware which is exactly the same except it's modified to look in that cookie instead of the `Authorization` header.

3) On my route handlers, I use the `RequireRecordAuth` middleware like

```go
e.POST("/my-route", myHandler, apis.RequireRecordAuth("users"))
```

Seems to work. Does this sound about right security-wise? Thanks for any help and happy to give more info.

**Comment by ganigeorgiev** [comment, marked as answer]
I think it is OK but I'd recommend to specify also the `Secure`, `MaxAge` and `SameSite` cookie options. Something like:

```go
&http.Cookie{
    Name:     "Auth-Token",
    Value:    t,
    HttpOnly: true,
    Path:     "/",
    Secure:   !app.IsDev(),
    MaxAge:   int(app.Settings().RecordAuthToken.Duration),
    SameSite: http.SameSiteStrictMode,
}
```

**Comment by 0ui** [comment]
Ah, yes I'll add those too. Perfect, thanks for the help!
